src/Ocelot/RateLimit/Middleware/ClientRateLimitMiddleware.cs

```diff
@@ -17,14 +17,17 @@ public class ClientRateLimitMiddleware : OcelotMiddleware
     {
         private readonly RequestDelegate _next;
         private readonly ClientRateLimitProcessor _processor;
+        private IHttpContextAccessor _httpContextAccessor;
 
         public ClientRateLimitMiddleware(RequestDelegate next,
             IOcelotLoggerFactory loggerFactory,
-            IRateLimitCounterHandler counterHandler)
+            IRateLimitCounterHandler counterHandler, 
+            IHttpContextAccessor httpContextAccessor)
```

Comment:
I would either:
- Ensure this service is in the DI container. Perhaps through the `AddOcelot` method (`TryAddScoped`), or
- Document the dependency, and silently ignore the logic if the service does does not exist in the DI container

Reply:
Yes, we will!
Thanks pointing to the issue!
Sometimes contributors open PRs with draft solutions. So, a multiple reviews are required.

One note, if there is no service in DI, then DI container should generate error. Am I wrong?
So, Ocelot app won't start.
Anyway, we will add safe check of this service existence in DI.

Reply:
Yeah, that is another option: to fail if the service is not in he DI... 
Anyway, as you inject the service in the constructor, the middleware class can not be instantiated by the DI container if the service can not be resolved, and an exception will be thrown

Reply:
Ok, I have removed the dependency from the constructor.

Reply:
Fixed in commit https://github.com/ThreeMammals/Ocelot/pull/1307/commits/03db4c591d8bf48a9976383df5f2ecf559f8e69c